src/platform_impl/web/web_sys/canvas.rs

```diff
@@ -73,22 +75,16 @@ impl Canvas {
                 .map_err(|_| os_error!(OsError("Failed to set a tabindex".to_owned())))?;
         }
 
-        let size = attr
-            .inner_size
-            .unwrap_or(
-                LogicalSize {
-                    width: 1024.0,
-                    height: 768.0,
-                }
-                .into(),
-            )
-            .to_physical(super::scale_factor(&window));
+        if let Some(size) = attr.inner_size {
+            let size = size.to_physical(super::scale_factor(&window));
+            super::set_canvas_size(&window, &canvas, size);
+        }
 
-        let canvas = Canvas {
+        Ok(Canvas {
             common: Common {
                 window,
                 raw: canvas,
-                size: Rc::new(Cell::new(size)),
+                size: Rc::default(),
```

Comment:
Just to double-check: you're aware that doing this means that `Window::inner_size` will return `(0, 0)` until the first `ResizeObserver` callback happens, usually after `EventLoop::run` is called? I'm not necessarily against that change but it should probably be documented, since it's a pretty important caveat to know about.

This used to be a bigger problem because `wgpu` would panic when you passed it a canvas with a framebuffer size of 0, but that's no longer a issue since winit no longer sets that. That's why I'm now fine for it to work this way.

Reply:
I was aware. Though this isn't new behavior to Winit, just new to Web. So users of `wgpu` had to check for zero-sized windows already (I encountered this myself as well on X11). Though looking at the documentation I can see no mention of it anywhere, so it would probably be good to document this somewhere.

This is probably off-topic for this PR, but we might want to consider a fix for this, like returning `None` if the Window doesn't have a size yet. This would be good for Web as well because if the canvas isn't inserted into the DOM it doesn't have a size either.

Reply:
Tracking this in https://github.com/rust-windowing/winit/issues/2863.